sdk/tests/conformance/glsl/misc/sampler-operand.html

```diff
@@ -0,0 +1,114 @@
+** permit persons to whom the Materials are furnished to do so, subject to
+** the following conditions:
+**
+** The above copyright notice and this permission notice shall be included
+** in all copies or substantial portions of the Materials.
+**
+** THE MATERIALS ARE PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+** EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+** MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+** IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+** CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+** TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+** MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
+*/
+
+-->
+
+<!DOCTYPE html>
+<html>
+<head>
+<meta charset="utf-8">
+<title>WebGL GLSL Conformance Tests - sampler operands</title>
+<link rel="stylesheet" href="../../../resources/js-test-style.css"/>
+<link rel="stylesheet" href="../../../resources/glsl-feature-tests.css"/>
+<script src="../../../js/js-test-pre.js"></script>
+<script src="../../../js/webgl-test-utils.js"></script>
+<script src="../../../js/glsl-conformance-test.js"></script>
+</head>
+<body>
+<div id="description"></div>
+<div id="console"></div>
+
+<script>
+// ESSL(1.00, 3.00, 3.10) section 4.1.7
+// Except for array indexing, structure field selection, and
+// parentheses, samplers are not allowed to be operands in expressions.
+</script>
+
+<script id="samplerArrayIndexing" type="x-shader/x-fragment">
+uniform sampler2D s2D[2];
+void main() {
+    gl_FragColor = texture2D(s2D[1], vec2(0.5, 0.5));
+}
+</script>
+
+<script id="samplerMemberSelection" type="x-shader/x-fragment">
+uniform struct S {
+    sampler2D a;
+} s;
+void main() {
+    gl_FragColor = texture2D(s.a, vec2(0.5, 0.5));
+}
+</script>
+
+<script id="samplerInParentheses" type="x-shader/x-fragment">
+uniform sampler2D s2D;
+void main() {
+    gl_FragColor = texture2D((s2D), vec2(0.5, 0.5));
+}
+</script>
+
+<script id="samplerAdding" type="x-shader/x-fragment">
+// This covers an ANGLE bug.
+// See https://bugs.chromium.org/p/angleproject/issues/detail?id=2028.
+uniform sampler2D s1;
+uniform sampler2D s2;
+void main() {
+    gl_FragColor = texture2D(s1 + s2, vec2(0.5, 0.5));
+}
+</script>
+
+<script>
+"use strict";
+GLSLConformanceTester.runTests([
+    { fShaderId: 'samplerArrayIndexing',
+      fShaderSuccess: true,
+      linkSuccess: true,
+      passMsg: 'array indexing on samplers should succeed'
+    },
+
+    { fShaderId: 'samplerMemberSelection',
+      fShaderSuccess: true,
+      linkSuccess: true,
+      passMsg: 'Member selection on samplers should succeed'
+    },
+
+    { fShaderId: 'samplerInParentheses',
+      fShaderSuccess: true,
+      linkSuccess: true,
+      passMsg: 'Samplers in parentheses should succeed'
+    },
+
+    // ANGLE bug.
+    { fShaderId: 'samplerAdding',
+      fShaderSuccess: false,
+      linkSuccess: false,
```

Comment:
You check the linkStatus here. But I suppose that the compile status should fail (before linkage), according to the ANGLE bug at https://bugs.chromium.org/p/angleproject/issues/detail?id=2028.